plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,680 @@
+	RequestsCount               int32   `json:"RequestsCount"`
+	RequestAverageDuration      float64 `json:"RequestAverageDuration"`
+}
+
+type databaseIndexesMetrics struct {
+	Count         int64 `json:"Count"`
+	StaleCount    int32 `json:"StaleCount"`
+	ErrorsCount   int64 `json:"ErrorsCount"`
+	StaticCount   int32 `json:"StaticCount"`
+	AutoCount     int32 `json:"AutoCount"`
+	IdleCount     int32 `json:"IdleCount"`
+	DisabledCount int32 `json:"DisabledCount"`
+	ErroredCount  int32 `json:"ErroredCount"`
+}
+
+type databaseStorageMetrics struct {
+	DocumentsAllocatedDataFileInMb int64 `json:"DocumentsAllocatedDataFileInMb"`
+	DocumentsUsedDataFileInMb      int64 `json:"DocumentsUsedDataFileInMb"`
+	IndexesAllocatedDataFileInMb   int64 `json:"IndexesAllocatedDataFileInMb"`
+	IndexesUsedDataFileInMb        int64 `json:"IndexesUsedDataFileInMb"`
+	TotalAllocatedStorageFileInMb  int64 `json:"TotalAllocatedStorageFileInMb"`
+	TotalFreeSpaceInMb             int64 `json:"TotalFreeSpaceInMb"`
+}
+
+type indexesMetricResponse struct {
+	Results         []*indexMetrics `json:"Results"`
+	PublicServerUrl string          `json:"PublicServerUrl"`
+	NodeTag         string          `json:"NodeTag"`
+}
+
+type indexMetrics struct {
+	DatabaseName               string   `json:"DatabaseName"`
+	IndexName                  string   `json:"IndexName"`
+	Priority                   string   `json:"Priority"`
+	State                      string   `json:"State"`
+	Errors                     int32    `json:"Errors"`
+	TimeSinceLastQueryInSec    *float64 `json:"TimeSinceLastQueryInSec"`
+	TimeSinceLastIndexingInSec *float64 `json:"TimeSinceLastIndexingInSec"`
+	LockMode                   string   `json:"LockMode"`
+	IsInvalid                  bool     `json:"IsInvalid"`
+	Status                     string   `json:"Status"`
+	MappedPerSec               float64  `json:"MappedPerSec"`
+	ReducedPerSec              float64  `json:"ReducedPerSec"`
+	Type                       string   `json:"Type"`
+	EntriesCount               int32    `json:"EntriesCount"`
+}
+
+type collectionsMetricResponse struct {
+	Results         []*collectionMetrics `json:"Results"`
+	PublicServerUrl string               `json:"PublicServerUrl"`
+	NodeTag         string               `json:"NodeTag"`
+}
+
+type collectionMetrics struct {
+	DatabaseName          string `json:"DatabaseName"`
+	CollectionName        string `json:"CollectionName"`
+	DocumentsCount        int64  `json:"DocumentsCount"`
+	TotalSizeInBytes      int64  `json:"TotalSizeInBytes"`
+	DocumentsSizeInBytes  int64  `json:"DocumentsSizeInBytes"`
+	TombstonesSizeInBytes int64  `json:"TombstonesSizeInBytes"`
+	RevisionsSizeInBytes  int64  `json:"RevisionsSizeInBytes"`
+}
+
+var sampleConfig = `
+  ## Node URL and port that RavenDB is listening on.
+  url = "https://localhost:8080"
+
+  ## RavenDB X509 client certificate setup
+  tls_cert = "/etc/telegraf/raven.crt"
+  tls_key = "/etc/telegraf/raven.key"
```

Comment:
Are those mandatory or do you also support non-TLS connections?

Reply:
we support both secured and non-secured servers, so I prepend this with `#`.